lib/src/local_csv/local_csv_generator.dart

```diff
@@ -0,0 +1,263 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:csv/csv.dart';
+import 'package:intl/intl.dart';
+import 'package:sheet_loader_localization/src/local_csv/local_csv_localization.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:path/path.dart' as path;
+
+class LocalCsvGenerator extends GeneratorForAnnotation<LocalCsvLocalization> {
+  @override
+  FutureOr<String> generateForAnnotatedElement(
+      Element element, ConstantReader annotation, BuildStep buildStep) async {
+    try {
+      final current = Directory.current;
+
+      final sourceDir = annotation.read('path').stringValue;
+      final sourceFile = annotation.read('sourceFile').stringValue;
+
+      final source = Directory.fromUri(Uri.parse(sourceDir));
+      final sourcePath = Directory(path.join(current.path, source.path));
+
+      if (!await sourcePath.exists()) {
+        String errorMessage = 'Source path does not exist';
+
+        stderr.writeln(errorMessage);
+        throw Exception(errorMessage);
+      }
+
+      var files = await dirContents(sourcePath);
+      if (sourceFile.isNotEmpty) {
+        final seekSourceFile = File(path.join(source.path, sourceFile));
+        if (!await seekSourceFile.exists()) {
+          throw ('Source file does not exist (${sourceFile.toString()})');
+        }
+        files = [seekSourceFile];
+      } else {
+        //filtering format
+        files = files.where((f) => f.path.contains('.csv')).toList();
+      }
+
+      var result = '';
+      if (files.isNotEmpty) {
+        result = await _generateFile(
+            element: element,
+            annotation: annotation,
+            files: files,
+            currentPath: current.path) ?? '';
+      } else {
+        throw(Exception('Source path empty'));
+      }
+      return result;
+    } catch (e) {
+      throw Exception('File generation error $e');
+    }
+  }
+
+  FutureOr<String?> _generateFile({
```

Comment:
Can we extract this into a class and reuse it for both local and sheet generators?